File: Frontend/teama-watchstore-frontend/WatchStoreApp/src/app/interfaces/order-dto.ts
import { AddressDto } from "./address";
import { UserDto } from "./user-dto";
import watchDto from "./watchDto";

export interface OrderDto {
    id:string;
    amount:number;
    orderStatus:string;
    quantity:number;
    timestamp:Date;
    statusTimestamp:Date;
    user:UserDto;
    watch:watchDto;
    deliveryAddress:AddressDto;
}

File: teama-watchstore/watchstore/watchstore-controller/src/main/java/com/nagarro/watchstore/dto/OrderDto.java
package com.nagarro.watchstore.dto;

import java.math.BigDecimal;
import java.time.LocalDateTime;

import lombok.Data;

/**
 * The OrderDto class represents the DTO (Data Transfer Object) for an order.
 * It contains the order details such as ID, quantity, status, amount, timestamps,
 * user details, delivery address, and watch details.
 * 
 * @author tushar01
 */
@Data
public class OrderDto {

	private String id;
	private int quantity;
	private OrderStatusDto orderStatus;
	private BigDecimal amount;
	private LocalDateTime timestamp;
	private LocalDateTime statusTimestamp;
	private UserProfileDto user;
	private AddressDto deliveryAddress;
	private WatchDto watch;
}
